### ISSUE TITLE
LOW: [$500] Selection List Refactor: Simple Selection List

### ISSUE THREAD

**Issue by thiagobrez** [Expensify/App#20354]
This issue keeps track of Phase 3 of the Selection List Refactor, in which we are refactoring all different list component variations into 3 new, clean components:

- Phase 1: Radio Button List
- Phase 2: Checkbox List
- Phase 3: Simple Selection List (this phase)

Thoroughly discussed in the parent issue: https://github.com/Expensify/App/issues/11795#issuecomment-1573880715
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a737edd58e0c740d</li>
        <li>Upwork Job ID: 1698702807486967808</li>
        <li>Last Price Increase: 2023-09-04</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @cristipaval</details>

**Comment by thiagobrez** [comment]
I imagine I will be able to start on this next week.

- Phase 1 PR is in final stages of review
- Phase 2 is almost ready. Must wait Phase 1 PR to be merged before raising this PR
- Then, Phase 3 (this one) can be started

**Comment by cristipaval** [comment]
I forgot to assign you to the issue, @thiagobrez . Sorry for that

**Comment by cristipaval** [comment]
I'll set this one on hold until the second phase is done.

**Comment by thiagobrez** [comment]
Starting to investigate this today, while second phase PR is in review

**Comment by thiagobrez** [comment]
Good news, these lists are fitting well with the new components. Group members list is ready

**Comment by thiagobrez** [comment]
Update: New Chat list is ready. Will be OOO for the rest of the week, so I'll come back to this on Monday

**Comment by thiagobrez** [comment]
Hi @shawnborton!

Context:

This issue is Phase 3 of the lists refactor (simple lists, no selection). One of these lists if the New Chat list (image below) ⬇️ 

![Screenshot 2023-07-25 at 14 19 53](https://github.com/Expensify/App/assets/26878038/64611f06-3576-45c7-8bae-50e7f7712b7c)

Question:

This list is being reused for both New Chat and New Group lists. So I was wondering if maybe you have an update if I can proceed with the New Group list? Last time it was blocked ([here](https://github.com/Expensify/App/issues/20353#issuecomment-1602670230)) because you still needed to investigate the patterns around checkbox lists.

If there is not a decision on the New Group list yet, no problem, I can make it just for the New Chat list. I'm just checking because if I could change both at the same time, it would make it simpler :)

Thanks!

**Comment by thiagobrez** [comment]
Perfect, thanks for the quick response!

**Comment by thiagobrez** [comment]
Update: Phase 3 is almost finished, will probably already raise the PR once Phase 2 is merged.

All previously investigated lists by Arek https://github.com/Expensify/App/issues/11795#issuecomment-1543877225 were already replaced (with one exception: `SearchPage`), but I see some lists were missed from the initial investigation (regarding all phases), so I will be adding those here as well.

1. `SearchPage` is currently being migrated to functional component, so I'm waiting [this PR](https://github.com/Expensify/App/pull/23076) to be merged first before replacing the list

2. Lists in these pages were missed from the initial investigation, so I will be replacing them in this Phase:
- `NotificationsPreferencePage`
- `WriteCapabilityPage`
- `TaskAssigneeSelectorModal`
- `TaskShareDestinationSelectorModal`
- `MoneyRequestConfirmationList`
- `IOUCurrencySelection`
- `MoneyRequestParticipantsSplitSelector`

**Comment by thiagobrez** [comment]
Update: Phase 3 is almost finished, will probably already raise the PR once Phase 2 is merged.

**Comment by thiagobrez** [comment]
Update: Phase 3 is almost finished, will probably already raise the PR once Phase 2 is merged.

**Comment by thiagobrez** [comment]
@cristipaval Actually, Phase 3 was already in the works while Phase 2 was being reviewed, it's in a local branch. Next week I'll split it in smaller PRs as agreed [in the thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1692952886670059)

**Comment by cristipaval** [comment]
Great! Thank you @thiagobrez!

**Comment by cristipaval** [comment]
This is in progress, Melv!

**Comment by cristipaval** [comment]
no need to create new issues for each. Just make sure you add the PR scope in the description body.

**Comment by cristipaval** [comment]
I think this issue is not on hold anymore, is it?

**Comment by cristipaval** [comment]
Added `External` label to have a bug zero and a c+ assigned.

**Comment by thiagobrez** [comment]
Hey @rushatgabhane , yeah I will be sending a few PRs here. Just FYI that most of Callstack is off this week due to RNEU. I'll get back to this on Monday!

**Comment by thiagobrez** [comment]
Update: We're progressing here. I'll send a few small focused PRs here to review more easily. First one is regarding [this issue](https://github.com/Expensify/App/issues/12668), in which I posted a question before I can proceed

**Comment by thiagobrez** [comment]
Holding on New Chat and New Group list replacements due to the new Global create feature that will come soon, as discussed here: https://github.com/Expensify/App/issues/12668#issuecomment-1714039010

**Comment by thiagobrez** [comment]
First PR is being reviewed here, regarding selection highlights: https://github.com/Expensify/App/pull/27246

**Comment by thiagobrez** [comment]
This is being worked on, planning to have another PR ready today or tomorrow

**Comment by thiagobrez** [comment]
Second PR is ready for review here: https://github.com/Expensify/App/pull/27767

No lists are replaced in this PR. This implements a base for the next pages to use

**Comment by thiagobrez** [comment]
Update - Phase 3 PRs:

[Reviewing] Selection highlight: https://github.com/Expensify/App/pull/27246
[Reviewing] Base implementation: https://github.com/Expensify/App/pull/27767
[Reviewing] Workspace Settings - Currency selector push-to-page: https://github.com/Expensify/App/pull/27861

Heads up: I'll be OOO from tomorrow until Tuesday - if anything urgent comes up, someone from Callstack can pick it up!

**Comment by thiagobrez** [comment]
Update - Phase 3 PRs:

[MERGED] ~~Selection highlight: https://github.com/Expensify/App/pull/27246~~
[REVIEWING] Base implementation: https://github.com/Expensify/App/pull/27767
[ON HOLD] Workspace Settings - Currency selector push-to-page: https://github.com/Expensify/App/pull/27861

**Comment by thiagobrez** [comment]
Update - Phase 3 PRs:

[MERGED] Selection highlight: https://github.com/Expensify/App/pull/27246
[MERGED] Base implementation: https://github.com/Expensify/App/pull/27767
[REVIEWING] Workspace Settings - Currency selector push-to-page: https://github.com/Expensify/App/pull/27861

**Comment by JmillsExpensify** [comment]
Looking at the linked PRs, it looks like the payment summary is:

 - Reviewer: @allroundexperts $250
 - Reviewer: @abdulrahuman5196 $250

@rushatgabhane did I miss any and you ended up reviewing a linked PR?

**Comment by thiagobrez** [comment]
@JmillsExpensify @rushatgabhane This is not done yet, there will be more incremental PRs coming in here, as agreed in the comments above

cc @cristipaval

**Comment by cristipaval** [comment]
Making this one `Weekly` as it is the parent issue for multiple PRs

**Comment by lukemorawski** [comment]
hey, it's Lucas from Callstack. I'm taking this over from Thiago

**Comment by cristipaval** [comment]
Thanks @lukemorawski !

**Comment by JmillsExpensify** [comment]
Still being worked on

**Comment by cristipaval** [comment]
What's the status here, @lukemorawski ?

**Comment by lukemorawski** [comment]
@cristipaval refactoring `<MoneyRequestParticipantsSelector />` list. Should have a PR tomorrow

**Comment by rushatgabhane** [comment]
@lukemorawski do we have a PR? Please link it here if so, thanks!

**Comment by lukemorawski** [comment]
@rushatgabhane hey, sorry had some urgent issues to close. will be posting a PR today

**Comment by cristipaval** [comment]
Thanks @lukemorawski

**Comment by rushatgabhane** [comment]
@lukemorawski gentle bump :)

**Comment by lukemorawski** [comment]
@rushatgabhane i'm finally free from other ticketa, so this has my full attention!

**Comment by lukemorawski** [comment]
ah, sorry but had to switch again to a more urgent task (theme switching). Back on it ASAP.

**Comment by JmillsExpensify** [comment]
Still working through what remains.

**Comment by JmillsExpensify** [comment]
Hmmm, like I said Melvin, we're still working through it.

**Comment by lukemorawski** [comment]
i'm returning to this ticket today

**Comment by lukemorawski** [comment]
@cristipaval I have refactored the list component in the `MoneyRequestParticipantsPage` component. But before I post a PR just wanted to clarify some stuff, mainly the way user can choose items from the list to make a split payment or just go directly to money request with single item from the list.
Currently, in the old list implementation, to make split you need press the little `Split` button present on every list item. If you want to just make a money request again a single user you just press on the list item in some other area then the `Split` button.
New list implementation, in it's multi-selection mode, present checkboxes on the left side of every list item. I have used those to enable splitting. It's kinda similar as before, but no explicit label saying "Split". If you want to make a simple money request, you just press on the list item, just like in the old version.
Is that new approach OK, or should I add the little "Split" button?
I'm guessing the reason those new list components where made, was to streamline UX and make it more consistent across the app. One thing that might not be clear with that new approach for a user is the fact it you can either select the row by tapping on the checkbox or just tap on the rest of the list item and call a different action.
What do you think. Oh, here's a video showing how it works now:


https://github.com/Expensify/App/assets/134388952/69a889d8-fb14-4d92-832d-0eda03255303

**Comment by cristipaval** [comment]
@Expensify/design could you please respond to the previous comment?

**Comment by shawnborton** [comment]
I thought the latest thinking was to re-add a "Split" option in the global create menu. Once we have that split-specific flow, I could see where the left checkboxes make sense but they also feel a bit heavy for mobile in this kind of flow, and I'd be fine just keeping things as they previously were

**Comment by trjExpensify** [comment]
>I didn't realize the list refactoring would touch the payment flows. cc @JmillsExpensify and @trjExpensify for thoughts here too.

Oh, I don't recall a discussion for switching radio buttons for checkboxes in the money request (and start chat?) flows. Especially as the multi-select is only applicable if you explicitly decide to split or create a group? 🤔

**Comment by cristipaval** [comment]
@shawnborton @trjExpensify I think this confusion happened because this issue is part of a massive refactoring (tracker [here](https://github.com/Expensify/App/issues/11795#issuecomment-1573880715)) we started a long time before adding the `Split` button in the `MoneyRequestParticipantsPage`.

To respond to @lukemorawski's question, we want the `Split` button exactly as it currently is, no replacement for the checkboxes, correct?

**Comment by shawnborton** [comment]
> To respond to @lukemorawski's question, we want the Split button exactly as it currently is, no replacement for the checkboxes, correct?

Yeah, I think we just want to leave this flow unchanged as I think we'll be following up and changing it soon anyways.

**Comment by lukemorawski** [comment]
@shawnborton @cristipaval great thanks, putting the Split button to the new list item

**Comment by JmillsExpensify** [comment]
Agree with leaving flow unchanged for now.

**Comment by JmillsExpensify** [comment]
Still working through implementation

**Comment by lukemorawski** [comment]
> Noticed another crash [#32692 (comment)](https://github.com/Expensify/App/pull/32692#discussion_r1453423290). cc: @lukemorawski @rushatgabhane @cristipaval @JmillsExpensify

PR is out for that https://github.com/Expensify/App/pull/34587

**Comment by JmillsExpensify** [comment]
Are good to pay out @rushatgabhane at this point? Also it looks like we might have had regressions?

**Comment by JmillsExpensify** [comment]
Ok cool. I'm going to remove the payment holds. Please let me know when we're done which linked PRs require compensation.

**Comment by JmillsExpensify** [comment]
@rushatgabhane please send a request for this in Expensify, and also don't forget the recommendation about whether a regression test is required. Closing this issue for now, and I'll circle back on it for payment.

**Comment by JmillsExpensify** [comment]
Opening this back up again since we're still waiting on some PRs.

**Comment by rushatgabhane** [comment]
PRs so far
- https://github.com/Expensify/App/pull/32692
- https://github.com/Expensify/App/pull/35841
- https://github.com/Expensify/App/pull/33080
- https://github.com/Expensify/App/pull/34978
- https://github.com/Expensify/App/pull/34356
- https://github.com/Expensify/App/pull/35413
- https://github.com/Expensify/App/pull/35567

**Comment by cristipaval** [comment]
We still have 2 open PRs for this

**Comment by JmillsExpensify** [comment]
Let me know when we're finally ready for payment on this one.

**Comment by JmillsExpensify** [comment]
Also @cristipaval you might need to comment on this issue to remove the `Overdue` label?

**Comment by JmillsExpensify** [comment]
Still not ready for payment, I believe, so removing that form the title again.

**Comment by cristipaval** [comment]
@lukemorawski the [refactoring](https://github.com/Expensify/App/pull/34356) for the NewChat page has been reverted. Will you open a new PR for it?

**Comment by lukemorawski** [comment]
@cristipaval yep, comming up in few minutes actually :)

**Comment by lukemorawski** [comment]
@cristipaval new PR here https://github.com/Expensify/App/pull/38610

**Comment by cristipaval** [comment]
> @cristipaval new PR here #38610

Thanks a lot! Could you please resolve the conflicts before @rushatgabhane starts reviewing?

**Comment by rushatgabhane** [comment]
Last PR to go! 🚀

**Comment by quinthar** [comment]
Should this be closed?  If not, what is the next step, who is doing it, and when?

**Comment by MrMuzyk** [comment]
@quinthar Hey, so we have a [PR](https://github.com/Expensify/App/pull/40785) that is waiting to be reviewed and tested again. After this is merged then whole refactor is complete and this issue can be closed.

Note: PR has been opened for 3 weeks now but it was hanging due to merge freeze I guess. In the meantime a new feature has been merged (uneven splits) that did some major changes in the refactored component and added a new feature to deprecated component that was being removed in this PR. I had to start pretty much from scratch thus the request to review and test it again before merging.

**Comment by rushatgabhane** [comment]
yep, it was reviewed 2 weeks ago but then a new feature was added in the deprecated component. Need to re-review now

**Comment by JmillsExpensify** [comment]
Can we close this issue out yet?

**Comment by MrMuzyk** [comment]
Refactor is complete. I'm fine closing it.

**Comment by laurenreidexpensify** [comment]
## Payment Summary:

### C+ Review

 @rushatgabhane 

- see https://github.com/Expensify/App/issues/20354#issuecomment-2081647087
    * 8 * PRs @ $500 (before price change) 
    * 2 * PRS @ $250 (after price change) 

Total: $4500

Payment Method: Please request in Newdot

### Contributor

@lukemorawski Agency contributor, no payment owed

**Comment by JmillsExpensify** [comment]
$4,5000 approved for @rushatgabhane
